### PR TITLE
Migrate GetState V1 to MigratingEndpointAdapter

### LINKED ISSUE #5538: Migrate GetState V1 to MigratingEndpointAdapter

### PR DESCRIPTION
**Pull request by courtneyeh**
fixes #5538 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/debug/GetStateIntegrationTest.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ public void shouldGetStateAsOctetStream() throws IOException {
 
   @Test
   public void shouldRejectAltairStateRequestsForJson() throws IOException {
-    startRestAPIAtGenesis(SpecMilestone.ALTAIR);
+    startRestAPIAtGenesis(SpecMilestone.ALTAIR); // TODO check, should it not handle ALTAIR?
     final Response response = get("head");
     assertThat(response.code()).isEqualTo(SC_BAD_REQUEST);
```
Comment (courtneyeh) [inline data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/debug/GetStateIntegrationTest.java]: Does this mean `GetState` should not handle ALTAIR? @rolfyone

Comment (rolfyone) [reply data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/debug/GetStateIntegrationTest.java]: Hrm... im less worried about v1... it'll handle altair but i've also just put up a PR to remove it from the spec...

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetState.java [hunk 2]
```diff
@@ -91,45 +118,35 @@ public GetState(final ChainDataProvider chainDataProvider, final JsonProvider js
...
+      builder.withType(
+          block -> schemaDefinitionCache.milestoneAtSlot(block.getSlot()).equals(milestone),
+          schemaDefinitionCache
+              .getSchemaDefinition(milestone)
+              .getBeaconStateSchema()
+              .getJsonTypeDefinition());
     }
-    return Optional.of(
-        jsonProvider.objectToJSON(new GetStateResponse(version, response.getData())));
+    return builder.build();
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetState.java]: should be able to use that metadata utility...

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/debug/GetStateIntegrationTest.java (hunk 1 above)
Comment (courtneyeh) [reply data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/debug/GetStateIntegrationTest.java]: This integration test fails, as it seems to want ALTAIR to give a bad request.

Comment (rolfyone) [reply data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/debug/GetStateIntegrationTest.java]: oh - so that's your answer :)

Comment (ajsutton) [reply data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/debug/GetStateIntegrationTest.java]: Not sure if this helps but we shouldn't be able to get an Altair state as JSON from this API but should be able to get Altair states as SSZ.

Comment (rolfyone) [reply data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/debug/GetStateIntegrationTest.java]: in that case it should only list the PHASE0 definition, wont allow anything else..

Comment (courtneyeh) [reply data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/debug/GetStateIntegrationTest.java]: Ah okay that makes things tricky

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetState.java [hunk 3]
```diff
@@ -138,7 +140,13 @@ public void handleRequest(RestApiRequest request) throws JsonProcessingException
         future.thenApply(
             maybeStateAndMetaData ->
                 maybeStateAndMetaData
-                    .map(stateAndMetaData -> AsyncApiResponse.respondOk(stateAndMetaData.getData()))
+                    .map(
+                        stateAndMetaData -> {
+                          if (stateAndMetaData.getMilestone() != SpecMilestone.PHASE0) {
+                            throw new BadRequestException("SpecMilestone not PHASE0");
+                          }
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetState.java]: this is the bit where we need to be careful, we can do this check, but we also need to check if our desired response is json before throwing this....

Comment (courtneyeh) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetState.java]: Is it not always json once it gets to that point?

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**